src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -626,12 +626,13 @@ public function getDate()
     /**
      * Sets the Date header.
      *
-     * @param \DateTime $date A \DateTime instance
+     * @param \DateTimeInterface $date A \DateTimeInterface instance
      *
      * @return Response
      */
-    public function setDate(\DateTime $date)
+    public function setDate(\DateTimeInterface $date)
```

Comment:
Another solution could be to deprecate extending this functions and make them final in 4.0 (i prefer this solution).